tasks/configure-topics.yml

```
@@ -22,13 +22,46 @@
   tags:
     - create
 
-- name: Kafka Topics | Create a new partitioned topic on Kafka
-  command: >
-    {{ kafka_bins }}/kafka-topics --zookeeper {{ random_zookeeper }}
-    {{ kafka_params | join(" ") }} --create --topic {{ item }}
-  run_once: yes
-  with_items: "{{ default_kafka_topics | difference(kafka_topics_result.stdout_lines) }}"
-  changed_when:
-    - default_kafka_topics | difference(kafka_topics_result.stdout_lines) | length > 0
-  tags:
-    - create
+- block:
+    - name: Kafka Topics | Create a new partitioned topic on Kafka
+      command: >
+        {{ kafka_bins }}/kafka-topics --zookeeper {{ random_zookeeper }}
+        {{ kafka_params | join(" ") }} --create --topic {{ topic }}
+      run_once: yes
+      loop: "{{ default_kafka_topics | map(attribute='name') | list | difference(kafka_topics_result.stdout_lines) }}"
+      loop_control:
+        loop_var: topic
+      changed_when:
+        - default_kafka_topics | difference(kafka_topics_result.stdout_lines) | length > 0
+      tags:
+        - create
+
+    - name: Kafka Topics | Set topic lifetime
+      command: >
+        {{ kafka_bins }}/kafka-topics --zookeeper {{ random_zookeeper }}
+        --alter --topic {{ topic.name }} --config retention.ms={{ topic.retention_ms }}
+      run_once: yes
+      loop: "{{ default_kafka_topics }}"
+      loop_control:
+        loop_var: topic
+      when: topic.retention_ms is defined
+      tags:
+        - create
+
+  when: default_kafka_topics.0 is mapping
```

```suggestion
  when:
     - default_kafka_topics | map(attribute='name') | list | difference(kafka_topics_result.stdout_lines) | length > 0
```
This can provide idempotence, the conditional is the same from the previous comment

---

It's also worth mentioning that we have a higher-level similar [conditional here](https://github.com/macunha1/confluent-kafka-role/blob/master/tasks/main.yml#L134)